packages/wdio-mocha-framework/src/index.ts

```diff
@@ -202,8 +203,13 @@ class MochaAdapter {
 
         switch (hookName) {
         case 'beforeSuite':
+            this._suiteStartDate = Date.now()
+        // eslint-disable-next-line no-fallthrough
         case 'afterSuite':
             params.payload = this._runner?.suite.suites[0]
+            if (params.payload instanceof Object && typeof (params.payload) === 'object') params.payload.duration =
+                params.payload.duration ||
+                (Date.now() - this._suiteStartDate)
```

Comment:
No need to check for instanceof here
```suggestion
            if (typeof params.payload === 'object') {
                params.payload.duration = params.payload.duration ||
                (Date.now() - this._suiteStartDate)
            }
```

Reply:
@christian-bromann I added that to ensure payload is not null as type of returns object for null

Reply:
@christian-bromann i removed the check all together